Automattic/crowdsignal-ui
# Add page preview context menu and a duplicate option

This patch adds an expandable context menu to page previews inside the editor, replacing the existing delete page button.  
The new menu offers an additional _'Duplicate'_ options which allows duplicating pages.

It's worth noting that since pages live 'above' the editor - our `useClientId()` hook wouldn't kick in until you visit both the original and the duplicated page. Hence it's necessary to manually regenerate all `attributes.clientId` values for our blocks whenever a page is being duplicated.  
This isn't an issue for Gutenberg's block `clientId`s as they're only used internally by the editor itself and sort themselves out anytime the editor is reloaded.

![Screen Shot 2022-02-18 at 1 28 24 PM](https://user-images.githubusercontent.com/8056203/154683979-08a6140c-ab3e-40ac-bcde-693576bdfdc2.png)

Solves c/X11AH7af-tr.

# Testing

- Create a new project and proceed to add some content.
- Hovering over a page preview should display a button to trigger the new context menu.
- Try duplicating the page, verify that when the draft is saved any question on the duplicated pages have unique `attributes.clientId` values.
- Try deleting a page, it should work the same way as previously.
- The context menu isn't available for the confirmation page which can't be duplicated or deleted.
